### ISSUE TITLE
[Feature request] Integrating with SSH key agent

### ISSUE THREAD

**Issue by RMerl** [evsar3/sshfs-win-manager#27]
Integration with an SSH key agent (Putty's or another one) would allow for more secure keyfile handling.  The agent could take care of handling the password protection.

**Comment by rmunn** [comment]
Worth noting that there's an OpenSSH Authentication Agent service included in Windows 10, though it is disabled by default. But it's simple to enable and start it in the Services app, and it's guaranteed to be there in every Win10 installation. So if you add code to integrate with a key agent, please make sure the built-in Win10 OpenSSH agent is one of the ones you integrate with.

**Comment by evsar3** [comment]
Built-in Windows 10 SSH Agent uses a named pipe for communication instead of a socket as seen in Unix. I wasn't able to find any node package that implementa such communication method.
In order to make this work I guess I will have to write my own client, and this will be a hard work, since I need to study the protocol and implement in JavaScript.
I will also look up for other key management programs and see if they provide a better API.

**Comment by Luckyvb** [comment]
What about Putty Agent?

**Comment by lancetekk** [comment]
i recently started to look into it, no promises though :)

**Comment by lancetekk** [comment]
So, i figured out that using pageant is kind of easy. I will try to cover some corner cases and report back in

**Comment by lancetekk** [comment]
I am very sorry. While i thought it would be possible, it turns out that i oversaw an issue and i cant solve this right now. I will keep looking for a solution though.

**Comment by rmunn** [comment]
> Built-in Windows 10 SSH Agent uses a named pipe for communication instead of a socket as seen in Unix. I wasn't able to find any node package that implementa such communication method.

https://stackoverflow.com/a/32172145 shows an example of using Windows named pipes: basically, you would use `net.connect("\\\\.\\pipe\\INSERT_PIPE_NAME_HERE")` if that example is correct. (The pipe is `\\.\pipe\INSERT_PIPE_NAME_HERE` but the backslashes have to be backslash-escaped, of course). I haven't tested this myself, but it looks like it's possible.

**Comment by venatiodecorus** [comment]
+1 for this

**Comment by bviktor** [comment]
Oh yes, all my stuff uses my key loaded in Pageant. Windows Git client, WSL distros, PuTTY itself.... it'd be silly to save my passphrase in each and every SSH-related tool separately...

**Comment by itsTyrion** [comment]
this would be a great feature

**Comment by p45rJv** [comment]
> Oh yes, all my stuff uses my key loaded in Pageant. Windows Git client, WSL distros, PuTTY itself.... it'd be silly to save my passphrase in each and every SSH-related tool separately...

SSHFS-WIN has **NO** support for the MS port of OpenSSH server and client, its using cygwin ports. Same with putty, pageant. Thats why **they can share that ssh-agent.exe**.

Git, GNUPG and WSL2 (AF_UNIX proxy) has support for the MS port. **Thats why they can share that other ssh-agent.exe** 

Its not silly that you have to type your passhprase for each ssh-agent. SSHFS-WIN doesnt even support the MS port so its expected that the manager for SSHFS-WIN doesnt either.

**Comment by bviktor** [comment]
Yes, there’s no support atm, that’s precisely why this ticket exists. Thanks for your invaluable contribution to the topic.

**Comment by p45rJv** [comment]
Forgot to add that npiperelay could accomplish this. Point being: -1 for manager doing this, +1 for the app doing this :P

**Comment by shanto** [comment]
I think passing `-oIdentityAgent=SSH_AUTH_SOCK`, and omitting `-oPreferredAuthentications=password` to ssh_command will suffice. Documentation says, when SSH_AUTH_SOCK is passed as a string, the same environment variable is expanded.
